crates/wash/src/cli/update.rs

```diff
@@ -67,25 +67,105 @@ pub struct UpdateCommand {
     #[clap(long, short = 'd')]
     dry_run: bool,
 
-    // TODO(#30): support --major, --minor, --patch flags to control update granularity
     /// Point at a different repository for updates
     #[clap(long, default_value = REPO)]
     git: String,
 
     /// GitHub token for private repository access. Can also be set via GITHUB_TOKEN, GH_TOKEN, or GITHUB_ACCESS_TOKEN environment variables
     #[clap(long, env = "WASH_GITHUB_TOKEN")]
     token: Option<String>,
+
+    /// Allow major version updates (breaking changes)
+    #[clap(long, conflicts_with_all = ["minor", "patch"])]
+    major: bool,
+
+    /// Allow minor version updates (new features, no breaking changes)
+    #[clap(long, conflicts_with_all = ["major", "patch"])]
+    minor: bool,
+
+    /// Allow only patch updates (bug fixes only)
+    #[clap(long, conflicts_with_all = ["major", "minor"])]
+    patch: bool,
+}
+
+fn parse_version_tuple(tag: &str) -> Option<(u8, u8, u8)> {
```

Comment:
You may want to consider using the `semver` crate here, since that will prevent us from having to parse our own version 😄 The splitting logic to slice off the beginning `wash-v` is still good, though. 

Using the semver crate will handle pre-releases as well, e.g. `1.0.0-alpha.1`